### PR TITLE
[Structural Mechanics] add Damage 2D Constitutive Law

### PR DESCRIPTION
**Pull request by marandra**
This PR adds the 2D-version of the LinearIsotropicDamage3D constitutive law.

Only a couple of functions are overriding the 3D version. Most of the changes are for registration of the CL in Kratos.

A test is also included.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    : LinearIsotropicDamage3D(rOther)
+{
+}
+
+//********************************CLONE***********************************************
+//************************************************************************************
+
+ConstitutiveLaw::Pointer LinearIsotropicDamagePlaneStrain2D::Clone() const
+{
+    LinearIsotropicDamagePlaneStrain2D::Pointer pclone(new LinearIsotropicDamagePlaneStrain2D(*this));
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp]: kratos::make_shared

**Review by philbucher — COMMENTED** [review]
one minor comment

File: applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp (hunk 1 above)
Comment (marandra) [reply applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp]: True. I'll update the other CLs in an upcoming PR.

File: applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+LinearIsotropicDamagePlaneStrain2D::~LinearIsotropicDamagePlaneStrain2D() = default;
+
+//************************************************************************************
+//************************************************************************************
+
+void LinearIsotropicDamagePlaneStrain2D::CalculateConstitutiveMatrix(
+    Matrix &rConstitTensor,
+    const Properties &rMaterialProperties
+    )
+{
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp:52]: If this is the linear elastic constitutive matrix you can derive the law from the linear elastic one, in a future PR

**Review by loumalouomega — APPROVED** [review]
OK, just one minor point

**Comment by RiccardoRossi** [comment]
any hope you can also add a plane stress version? (not in this PR, only a suggestion)

File: applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp (hunk 2 above)
Comment (marandra) [reply applications/StructuralMechanicsApplication/custom_constitutive/linear_isotropic_damage_plane_strain_2d.cpp:52]: I'm preparing a followup PR with changes for this and other CLs. I will include this suggestion.

**Comment by marandra** [comment]
> any hope you can also add a plane stress version? (not in this PR, only a suggestion)

I don't currently have a stress version, but  we can discuss an implementation during the next meeting.

**Comment by marandra** [comment]
Thanks for the review

**Pull request merged**